packages/react/src/runtime/style-cache.tsx

```diff
@@ -12,13 +12,43 @@ const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | nu
 if (!isNodeEnvironment()) {
   /**
    * Iterates through all found style elements generated when server side rendering.
-   *
-   * @param cb
    */
   const ssrStyles = document.querySelectorAll<HTMLStyleElement>('style[data-cmpld]');
   for (let i = 0; i < ssrStyles.length; i++) {
+    // Create the client cache for all SSR'd classes
+    const stylesheet = ssrStyles[i];
```

Comment:
can you see if we can write some unit tests? looking at the tests there aren't any written, so it might just be a bit tricky to test this flow. let me know how you go 😄 

would be interested in these test cases:

1. hydrate single style
2. hydrate nested styles
3. hydrate ordered styles (bunch of pseudo classes, media queries, and regular CSS rules)

Reply:
Have added tests. Will add more to cover the rest of the cases above